fuse/data/ops/ops_read.py

```diff
@@ -113,3 +114,51 @@ def get_all_keys(self) -> List[Hashable]:
         :return: list of  dataframe index values
         """
         return list(self.data.keys())
+
+
+class OpReadHDF5(OpBase):
+    """
+    Op reading data from hd5f based dataset
+    """
+
+    def __init__(
+        self,
+        data_filename: Optional[str] = None,
+        columns_to_extract: Optional[List[str]] = None,
+        rename_columns: Optional[Dict[str, str]] = None,
+        key_index: str = "data.sample_id",
+        key_column: str = "sample_id",
+    ):
+        """
+        :param data_filename: path to hdf5 file
+        :param columns_to_extract: list of columns to extract - dataset keys to extract. When None (default) all columns are extracted
+        :param rename_columns: rename columns
+        :param key_index: name of value in sample_dict which will be used as the key/index
+        :param key_column: name of the column which use as key/index. In case of None, the original dataframe index will be used to extract the values for a single sample.
+        """
+        super().__init__()
+        # store input
+        self._data_filename = data_filename
+        self._columns_to_extract = columns_to_extract
+        self._rename_columns = rename_columns if rename_columns is not None else {}
+        self._key_index = key_index
+        self._key_column = key_column
+
+        self._h5 = h5py.File(self._data_filename, "r")
```

Comment:
Seeing that I was concern that we load the entire file into memory, but seems not:
https://stackoverflow.com/questions/40449659/does-h5py-read-the-whole-file-into-memory